codec.go

```diff
@@ -100,11 +102,40 @@ func (e *framedMsgpackEncoder) Encode(i interface{}) <-chan error {
 	select {
 	case <-e.doneCh:
 		ch <- io.EOF
+	case <-ctx.Done():
+		ch <- ctx.Err()
 	case e.writeCh <- writeBundle{bytes, ch}:
 	}
 	return ch
 }
 
+func (e *framedMsgpackEncoder) EncodeAndWriteAsync(ctx context.Context, i interface{}) <-chan error {
```

Comment:
The caller of this function just passes in context.Background(), so may as well remove the ctx parameter and the cases below